### PR TITLE
Fix a bug where we're iterating over a string, not an array

### PR DESCRIPTION
**Pull request by steelbrain**
Just like the title saysy

### CODE REVIEW

File: lib/helpers.coffee [hunk 1]
```diff
@@ -61,7 +61,7 @@ module.exports = Helpers =
       regex = xcache.get(rawRegex)
     else
       xcache.set(rawRegex, regex = XRegExp(rawRegex))
-    for line in data
+    for line in data.split(/\r?\n/g)
```
Comment (lee-dohm) [inline lib/helpers.coffee]: I don't believe the regular expression needs the `g` flag.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split#Removing_spaces_from_a_string

Comment (steelbrain) [inline lib/helpers.coffee]: I just removed the g flag and it works the same, Thanks! Gonna update.

**Comment by lee-dohm** [comment]
Looks ok to me ... don't know what bug this is intended to fix or if it actually fixes it though. Probably should get more feedback than just mine.

**Comment by k2b6s9j** [comment]
I just fixed the Travis Build. The PR build should reflect this if re-run.

**Comment by steelbrain** [comment]
Great :clap:

**Comment by steelbrain** [comment]
@lee-dohm The `Helpers::parse` method allows linter providers to parse messages according a regex, the input is a pure string, and we were doing
```js
for(let line in string)
```
instead of
```js
for(let line in string.split(/\r?\n/))
```
which was returning empty extractions array to providers

**Comment by k2b6s9j** [comment]
Actually, if you could merge master in that'd be awesome. I feel hesitant to merge this in if the existing tests don't pass.

**Comment by steelbrain** [comment]
The tests are green. :wink:

**Comment by k2b6s9j** [comment]
What happens when someone passes an array of strings?

**Comment by steelbrain** [comment]
You shouldn't pass an array of strings, You should foreach and pass them one by one instead

**Comment by k2b6s9j** [comment]
Alright. We should be good for now then. :shipit:

**Pull request merged**